### ISSUE TITLE
Node stuck after "Oct 17 09:53:34.001 INFO end of epoch, task: leadership"

### ISSUE THREAD

**Issue by mark-stopka** [input-output-hk/jormungandr#978]
**Describe the bug**
I have been running the node for about 12 hours, and last message I've seen in the log of the node was `Oct 17 09:53:34.001 INFO end of epoch, task: leadership`, that was more than 12 hours ago (approx, 24 hours since start of the node), my API reports following:

```
[mark_stopka@docker-host-01 jormungandr]$ ./jcli rest v0 node stats get -h http://127.0.0.1:3101/api

blockRecvCnt: 935
lastBlockDate: "236.16804"
lastBlockFees: 0
lastBlockHash: b5dc20133289e6f2f435d4fd13c25f6233e80f7abd505e617affa6495fdfa82b
lastBlockHeight: "131538"
lastBlockSum: 0
lastBlockTime: "2019-10-16T17:15:06+00:00"
lastBlockTx: 0
txRecvCnt: 0
uptime: 109482
```

**Mandatory Information**
1. `jcli --full-version` output;

```
[mark_stopka@docker-host-01 jormungandr]$ ./jcli --full-version
jcli 0.6.1 (HEAD-a0de8d4, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]
```

2. `jormungandr --full-version` output;

```
[mark_stopka@docker-host-01 jormungandr]$ ./jormungandr --full-version
jormungandr 0.6.1 (HEAD-a0de8d4, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]
```

**To Reproduce**
1. Run `RUST_BACKTRACE=full ./jormungandr --config ./node-config.yaml --genesis-block-hash adbdd5ede31637f6c9bad5c271eec0bc3d0cb9efb86a5b913bb55cba549d0770`
2. Wait...

```
Oct 16 23:29:17.609 INFO error connecting to peer, reason: Connection refused (os error 111), peer_addr: 54.153.19.202:3000, task: network
Oct 16 23:29:27.610 INFO error connecting to peer, reason: Connection refused (os error 111), peer_addr: 54.153.19.202:3000, task: network
Oct 16 23:29:37.610 INFO error connecting to peer, reason: Connection refused (os error 111), peer_addr: 54.153.19.202:3000, task: network
Oct 16 23:29:47.611 INFO error connecting to peer, reason: Connection refused (os error 111), peer_addr: 54.153.19.202:3000, task: network
Oct 16 23:29:57.609 INFO error connecting to peer, reason: Connection refused (os error 111), peer_addr: 54.153.19.202:3000, task: network
Oct 16 23:30:07.610 INFO error connecting to peer, reason: Connection refused (os error 111), peer_addr: 54.153.19.202:3000, task: network
Oct 16 23:30:27.610 INFO error connecting to peer, reason: Connection refused (os error 111), peer_addr: 54.153.19.202:3000, task: network
Oct 16 23:31:57.610 INFO error connecting to peer, reason: Connection refused (os error 111), peer_addr: 54.153.19.202:3000, task: network
Oct 17 09:53:34.001 INFO end of epoch, task: leadership
```

**Expected behavior**
Node will continue syncing with the network...

**Comment by mark-stopka** [comment]
I've just noticed I had a `storage` option in a config commented, so the node started up with `INFO storing blockchain in memory, task: init`, however if the node ran out of memory (which I haven't checked before restarting the process) I would expect a log message or a node crash (with an informative message nonetheless), besides, the VM on which the node was running has some 16GiB of memory, so it should not be the issue...

**Comment by mark-stopka** [comment]
Used `node-config.yaml`...
```
[mark_stopka@docker-host-01 jormungandr]$ cat node-config.yaml
log:
  format: plain
  level: info
  output: stderr
p2p:
  listen_address: /ip4/0.0.0.0/tcp/3000
  #public_address: /ip4/x.x.x.x/tcp/xxxx # NODE'S EXTERNAL IP ADDRESS AND PORT
  topics_of_interest:
    blocks: normal #change to high when running a stakepool
    messages: low #change to high when running a stakepool
  trusted_peers:
     - address: "/ip4/3.115.194.22/tcp/3000"
       id: ed25519_pk1npsal4j9p9nlfs0fsmfjyga9uqk5gcslyuvxy6pexxr0j34j83rsf98wl2
     - address: "/ip4/13.113.10.64/tcp/3000"
       id: ed25519_pk16pw2st5wgx4558c6temj8tzv0pqc37qqjpy53fstdyzwxaypveys3qcpfl
     - address: "/ip4/52.57.214.174/tcp/3000"
       id: ed25519_pk1v4cj0edgmp8f2m5gex85jglrs2ruvu4z7xgy8fvhr0ma2lmyhtyszxtejz
     - address: "/ip4/3.120.96.93/tcp/3000"
       id: ed25519_pk10gmg0zkxpuzkghxc39n3a646pdru6xc24rch987cgw7zq5pmytmszjdmvh
     - address: "/ip4/52.28.134.8/tcp/3000"
       id: ed25519_pk1unu66eej6h6uxv4j4e9crfarnm6jknmtx9eknvq5vzsqpq6a9vxqr78xrw
     - address: "/ip4/13.52.208.132/tcp/3000"
       id: ed25519_pk15ppd5xlg6tylamskqkxh4rzum26w9acph8gzg86w4dd9a88qpjms26g5q9
     - address: "/ip4/54.153.19.202/tcp/3000"
       id: ed25519_pk1j9nj2u0amlg28k27pw24hre0vtyp3ge0xhq6h9mxwqeur48u463s0crpfk
rest:
  listen: "127.0.0.1:3101"
explorer:
  enabled: false
#storage: ./tmp/jormungandr/ # create storage folder in <path> and update path below
mempool:
    fragment_ttl: 30m
    log_ttl: 30m
    garbage_collection_interval: 30m
```

**Comment by rdlrt** [comment]
Duplicate #956 and #972

**Comment by mark-stopka** [comment]
I am not sure if this was resolved with 0.6.5 but I am not hitting this bug anymore... As this is a duplicate, I am closing the issue.
